File: core/knowledge/entity_engine.py
from __future__ import annotations

from typing import Dict, List


def build_entities(symbols: List[str]) -> Dict[str, object]:
    unique = sorted(set(str(s) for s in (symbols or []) if s))
    return {"entities": [{"id": u, "kind": "symbol"} for u in unique], "count": len(unique)}

File: src/knowledge/entityEngine.ts
/**
 * Converted from Python: core/knowledge/entity_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function buildEntities(symbols: any): any {
  var unique: any = py.sorted(py.toSet(py.iter(py.or2(symbols, () => ([]))).filter((s: any) => py.truthy(s)).map((s: any) => py.toStr(s))));
  return {"entities": py.iter(unique).map((u: any) => ({"id": u, "kind": "symbol"})), "count": py.len(unique)};
}
